### DISCUSSION TITLE
Is there a shell version of this?

### DISCUSSION THREAD

**Discussion by busr1998** [mifi/lossless-cut#1205] [Ideas]
Hi, I was looking into running something like this from Terminal on Mac

Would be good to use a shell version

Thanks

**Comment by mifi** [comment]
llc has a few command line options, but it will probably never become a fully command line utility. Using ffmpeg directly is better for that. losslesscut does **everything** by calling ffmpeg, so if you need something automated, you can have a look at the "last ffmpeg commands" in the help page and the work from there.

**Comment by busr1998** [comment]
Thanks for your reply Mikael. Wow I didn't see that "last ffmpeg commands" button, it's great.

My previous code was:

`ffmpeg -ss 00:00:03 -i "$inputfile" -t 0:0:8 -avoid_negative_ts make_zero -vcodec copy -ar 44100 "$outputfile"`

and now, using the code from llc:

`ffmpeg -hide_banner -ss "3.00000" -i "$inputfile" -t "8.00000" -avoid_negative_ts make_zero -map "0:0" "-c:0" copy -map "0:1" "-c:1" copy -map_metadata 0 -default_mode infer_no_subs -ignore_unknown -f mp4 -y "$outputfile"`

I realise now that the reason it was not working was adding the sample rate -ar.  **Is it possible to change the sample rate of the audio without breaking?**

Regardless of that it does now work, so many thanks

**Reply by mifi** [reply]
not possible to change audio rate without re-encoding

**Reply by busr1998** [reply]
great, many thanks for confirming
